homeassistant/components/surepetcare/binary_sensor.py

```diff
@@ -0,0 +1,212 @@
+"""Support for Sure PetCare Flaps binary sensors."""
+import logging
+
+import homeassistant.helpers.device_registry as dr
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.const import CONF_ID, CONF_NAME, CONF_TYPE
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import (CONF_HOUSEHOLD_ID, DATA_SURE_PETCARE, DATA_SUREPY,
+                    DEFAULT_DEVICE_CLASS, DEFAULT_ICON, SURE_IDS, TOPIC_UPDATE,
+                    SureLocationID, SureLockStateID, SureProductID,
+                    SureThingID)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up Sure PetCare Flaps sensors based on a config entry."""
+    from surepy import SurePetcare
+
+    entities = list()
+    surepy: SurePetcare = hass.data[DATA_SURE_PETCARE][DATA_SUREPY]
+
+    for thing in hass.data[DATA_SURE_PETCARE][SURE_IDS]:
+        sure_id = thing[CONF_ID]
+        sure_type = thing[CONF_TYPE]
+
+        if sure_id not in hass.data[DATA_SURE_PETCARE][sure_type]:
+
+            if sure_type == SureThingID.FLAP.name:
+                hass.data[DATA_SURE_PETCARE][
+                    sure_type][sure_id] = await surepy.get_flap_data(sure_id)
+                entities.append(Flap(sure_id, thing[CONF_NAME], hass=hass))
+            elif sure_type == SureThingID.PET.name:
+                hass.data[DATA_SURE_PETCARE][
+                    sure_type][sure_id] = await surepy.get_pet_data(sure_id)
+                entities.append(Pet(sure_id, thing[CONF_NAME], hass=hass))
+
+    async_add_entities(entities, True)
+
+
+class SurePetcareBinarySensor(BinarySensorDevice):
+    """A binary sensor implementation for Sure Petcare Entities."""
+
+    def __init__(self, _id: int, name: str,
+                 icon=None, device_class=None, sure_type=None, hass=None):
+        """Initialize a Sure Petcare binary sensor."""
+        self._id = _id
+        self._hass = hass
+        self._name = name
+        self._unit_of_measurement = "%"
+        self._icon = icon
+        self._device_class = device_class
+
+        self._household_id = hass.data[DATA_SURE_PETCARE][CONF_HOUSEHOLD_ID]
```

Comment:
Pass in the id as a parameter to the entity instead.

Reply:
you mean self._id?! Why? That refers to the Sensor/Flap, not to the household.

Reply:
I mean the value that we store in `self._household_id`.